connman/src/main.c

```diff
@@ -532,6 +532,8 @@ static void parse_config(GKeyFile *config)
 		g_free(str);
 	}
 
+	g_clear_error(&error);
```

Comment:
1st thought: why not put this closer to where the error can end up being set?

2nd thought: after locating ^ place ->  the error value from ipv4 lookup is not used, and similar ipv6 lookup passes null error (in fact almost all lookups are like this) -> why not do that for ipv4 lookup too and thus have no dontcare error to clear here?

Reply:
> the error value from ipv4 lookup is not used, and similar ipv6 lookup passes null error...

Yup, it seems it just has been forgotten here in the ipv4 case https://github.com/sailfishos/connman/commit/519ced062e926f6d1ebd5adffa8c27c0fa8a52da#diff-56a8ba3b41201b3671fbec69b15cc878e34f61922921e4f821d79a7bf53c1c96R403

Reply:
Yeah, as the aim is to just use the default if the configuration is missing it might be best to drop the error from the IPv4 case as well. I'll force push the change.

Reply:
Done, dropped from IPv4 with 8af344097b142cacf064ead18dcebc8923ef6a20. Resolving this.